components/director/internal/domain/formation/service.go

```diff
@@ -159,14 +159,14 @@ func (s *service) CreateFormation(ctx context.Context, tnt string, formation mod
 			return nil, err
 		}
 		// TODO:: Currently we need to support both mechanisms of formation creation/deletion(through label definitions and Formations entity) for backwards compatibility
-		if err = s.createFormation(ctx, tnt, *templateName, formationName); err != nil {
+		if err = s.createFormation(ctx, tnt, templateName, formationName); err != nil {
```

Comment:
Can we remove lines 162-165? That way if everithing is fine with the labelDeffinition we will proceed with the code after the if statement. That way we will reduce the code duplication and the flow will be more straightforward.

Reply:
Currently the inner if(duplicated code) is needed because it cover the case when we have error that is `not found` and a few lines above we have `CreateWithFormations` method that creates the "old" formation and we need to double maintain(for time being) a formation entity as well. The outer if is when there is no errors from the `modifyFormations` method.

Reply:
If we do not return on line 165 we will proceed with the code on line 169 which is the same as line 162. If we remove lines 162-165 the flow will be - try to modify formations -> if notfound call create with formations and then create formation on line 169

Reply:
ah, to remove the return statement, yep, that will simplify the function. Initially I understand you to remove only the `s.createFormation` function